File: learn/atcxx/hash_map.cc
#include "./hash_map.h"

#include <iostream>

int main() {
  hashmap<int, char> m;

  m[1] = 'a';
  m[2] = 'b';
  m[9] = 'c';

  std::cout << "MAP: " << m << std::endl;

  return 0;
}

File: learn/algos/hash_map.py
def hash(key: str) -> int:
  return int(key) % 10


class HashMap:

  def __init__(self, nbuckets=10):
    self.buckets = dict((k, []) for k in range(nbuckets))

  def insert(self, key: str, value):
    hash_ = hash(key)
    if hash_ in self.buckets[hash_]:
      raise KeyError
    self.buckets[hash_].append((key, value))
